samples/java/reference-app/src/main/java/com/godaddy/asherah/referenceapp/App.java

```diff
@@ -58,12 +55,31 @@
 
   enum KmsType { STATIC, AWS }
 
+  @CommandLine.ArgGroup
+  private DynamoDbConfig dynamoDbConfig;
+
+  static class DynamoDbConfig {
+    @Option(names = "--dynamodb-endpoint", split = ",",
+        description = "Comma separated values for the dynamodb <service endpoint,signing region> " +
+        "(only supported by DYNAMODB)")
+    private static String[] dynamoDbEndpointConfig;
+    @Option(names = "--dynamodb-region",
+        description = "The AWS region for DynamoDB requests (only supported by DYNAMODB)")
+    private static String dynamoDbRegion;
+  }
+
   @Option(names = "--metastore-type", defaultValue = "MEMORY",
       description = "Type of metastore to use. Enum values: ${COMPLETION-CANDIDATES}")
   private MetastoreType metastoreType;
   @Option(names = "--jdbc-url",
       description = "JDBC URL to use for JDBC metastore. Required for JDBC metastore.")
   private String jdbcUrl;
+  @Option(names = "--enable-region-suffix",
+      description = "Configure the metastore to use regional suffixes (only supported by DYNAMODB)")
+  private String regionSuffix;
+  @Option(names = "--dynamodb-table-name",
+      description = "The table name for DynamoDb (only supported by DYNAMODB)")
+  private String tableName;
```

Comment:
The class `DynamoDbConfig` ensures that endpoint and region are mutually exclusive. Moving tableName to that class would require additional logic since that doesn't need to be mutually exclusive of the other two